### PR TITLE
Makefile: Drop hard-coded Bash and simplify gofmt

### PR DESCRIPTION
**Pull request by wking**
I'm reverting #870 here and also removing the code that seemed to have motivated it.  Details on the individual changes in the commit messages.  The `find` commit is something of a drive-by; I'm happy to split it out into its own pull request if that would make review easier.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -41,6 +39,10 @@ endif
 	@./scripts/verify-gopkg.sh
 	@go test -v $(shell go list ./... | grep -v '/vendor/\|/e2e')
 
+gofmt:
+	gofmt -s -w $(GOFILES)
```
Comment (ericchiang) [inline Makefile]: We might be able to replace this with just `go fmt ./...`

Comment (ericchiang) [reply Makefile]: It automatically does as of 1.9 :)

See https://golang.org/doc/go1.9#vendor-dotdotdot

Comment (wking) [reply Makefile]: > It automatically does as of 1.9 :)

Are we comfortable making that our minimum release, as documented [here][1]?  I see we dropped 1.8 from Travis in b64b287dfe643 (#919).

[1]: https://github.com/kubernetes-incubator/bootkube/blob/bff16e2afc34df7469e7a108cf5d87098809249a/Documentation/development.md#requirements

**Comment by ericchiang** [comment]
Yeah, let's bump it to 1.10 and add a comment to the travis file to update
that number when we're bumping the travis versions. This can be a separate
PR.

File: Makefile (hunk 1 above)
Comment (wking) [reply Makefile]: > We might be able to replace this with just `go fmt ./...`

I was taking another look at this now that #955 has bumped our minimum Go version.  `go fmt …` calls `gofmt` with `-l -w` by default, which means we'd be dropping the `-s` we're currently using.  Is that acceptable?  Or do you want to keep the current, direct `gofmt` invocation?

**Comment by wking** [comment]
While I wait for maintainer feedback about [whether we need to keep `-s`][1], I've rebased this onto master (around #949, #956) with 99b62c23 → 9815a948.

[1]: https://github.com/kubernetes-incubator/bootkube/pull/948#discussion_r178906706

**Pull request merged**